python/mp/rosary.py

```diff
@@ -144,42 +154,50 @@ def find_defined_effects(self, module_or_class):
         #   http://stackoverflow.com/a/22578562
         for name, obj in inspect.getmembers(module_or_class):
             if inspect.ismodule(obj) and obj.__package__ == 'mp.effects':
-                classes = classes.union(self.find_defined_effects(obj))
+                classes = classes.union(self.find_written_effects(obj))
             elif inspect.isclass(obj):
                 classes.add(obj)
 
         return classes
 
 
-    def register_defined_effects(self):
-        defined_effects = self.find_defined_effects(effects)
+    def register_written_effects(self):
+        """
+        Find all the effect that people have written and register them.
+        """
+
+        defined_effects = self.find_written_effects(effects)
         for eff in defined_effects:
             # Don't register abstract classes, e.g. effects.effect.Effect
             if not inspect.isabstract(eff) and issubclass(eff, effects.effect.Effect):
                 self.register_effect(eff)
 
 
     def register_trigger(self, trigger_class):
+        """
+        Unlike effects, instead of holding on to the init function,
+        just go ahead and instantiate the trigger once
 
-        print("REGISTER A TRIGGER")
-        print(trigger_class)
+        (Maybe reconsider this later)
+        """
 
         # Instantiate
-        self.trigger_id += 1
-        trigger = trigger_class()
-        trigger.rosary = self
-        trigger.id = self.trigger_id
-        self.triggers[trigger.name] = trigger
-
-        print("WHAT ARE MY TRIGGERS")
-        print(self.triggers)
-
-        self.dispatcher.map(trigger.osc_path,
-                            trigger.trigger_wrapper,
-                            trigger)
+#        self.trigger_id += 1
+#        trigger = trigger_class()
+#        trigger.rosary = self
+#        trigger.id = self.trigger_id
+#        self.triggers[trigger.name] = trigger
+#
+#        self.dispatcher.map(trigger.osc_path,
+#                            trigger.trigger_wrapper,
+#                            trigger)
+        print("REGISTER TRIGGER")
+        print(trigger_class)
+        t = trigger_class()
+        self.trigger_registry[t.name] = trigger_class
```

Comment:
So instead of mapping the trigger name to the instance method's `next()`, which looked at the `effect_sequence`, we're now going to create a `Trigger` object, and we're keeping a `self.trigger_registry` much like for effects, on how to make those instances.